### ISSUE TITLE
Zot Protocol Support

### ISSUE THREAD

**Issue by sunjam** [nextcloud/social#228]
Zot is a protocol being used to federate instances of [Hubzilla](https://framagit.org/hubzilla/core), Osada, Red, and Zap.  [See Osada for a social network that bridges between Zot and ActivityPub](https://git.framasoft.org/macgirvin/osada).  Zot's nomadic identity could also be [very useful to the Nextcloud core server as well](https://github.com/nextcloud/server/issues/11653).  Here is a link to the latest [Zot6 documentation](https://macgirvin.com/wiki/mike/Zot(20)VI/Home?f=&owt=b7b68662248656474b7d3c456c1da923).

**Comment by jancborchardt** [comment]
We prefer to use standards which are well-established and widely used already (like WebDAV, ActivityPub etc) and that’s what we focus on with our limited resources.

If anyone would like to implement this, feel free. But it’s not a main roadmap feature.

**Comment by m4sk1n** [comment]
@elvecio
they assumed that Diaspora is going to implement ActivityPub

**Comment by jancborchardt** [comment]
Yup, as @m4sk1n said – Diaspora and MediaGoblin seem to have plans to implement ActivityPub, that’s what the statement comes from. We think considering the limited resources of all of our projects it only makes sense if we all use a common standard.

**Comment by strypey** [comment]
Just to clarify, my understanding is that Diaspora are *not* planning to implement ActivityPub. The latest comment I've seen on the subject from any of the active developers is here:
https://schub.io/blog/2018/02/01/activitypub-one-protocol-to-rule-them-all.html

AFAIK the other apps that currently support the Diaspora protocol either already have implemented ActivityPub ([Hubzilla](https://framagit.org/hubzilla/addons/tree/master/pubcrawl)), or are working on it ([Friendica](https://friendi.ca/2018/11/18/activitypub-support-in-friendica/), [SocialHome](https://git.feneas.org/socialhome/socialhome/issues/522)). Not sure about GangGo, but it's [homepage mentions federation with Mastodon](https://ggg.social/), so I presume so.

**Comment by jancborchardt** [comment]
Right now we support ActivityPub as it's a good standard used by the biggest (as far as we know) decentralized social network, which is Mastodon.

Especially considering our limited resources, we don't have any plans to implement other protocols.

**Comment by elvecio** [comment]
_my understanding is that Diaspora are not planning to implement ActivityPub._

that's why  I asked ;)

**Comment by strypey** [comment]
On 2018-12-15 18:56, Jan-Christoph Borchardt wrote:
> Right now we support ActivityPub as it's a good standard used by the biggest (as far as we know) decentralized social network, which is Mastodon.

AP is certainly the most widely supported at present, with a large
number of apps already supporting it, others with implementations in the
works (as I mentioned above), and many new apps being developed to work
with it:
https://gitlab.com/fediverse/fediverse.gitlab.io/wikis/watchlist-for-activitypub-apps

> Especially considering our limited resources, we don't have any plans to implement other protocols.

Fair enough. Just pointing out that Zot solves a number of problems that
AP doesn't and is worth keeping a weather eye on for the future :)

**Comment by m4sk1n** [comment]
Maybe one day we will have an AP extension including normadic identities?

**Comment by strypey** [comment]
Marcin Mikołajczak wrote:
> Maybe one day we will have an AP extension including normadic identities?

Maybe, but from what little I do understand so far about the differences
between AP and Zot, I don't think that's likely. Mainly because of the
way identity in AP is strongly tried to domain names.

I think the most misunderstood point made in the famous XKCD comic about
standards, is that there are *good reasons* for having competing
standards, because no one standard can properly address every
conceivable use case. I've tried to lay out here the particular use
cases that I see the various existing federation standards being applied
to:
https://gitlab.com/fediverse/fediverse.gitlab.io/wikis/uses-of-various-federating-(and-some-non-federating)-protocols


The other important point about standards is what they standardize,
which is mainly about which *protocols* to use for which *functions*,
and how to use them. If there was ever a universal federation standard,
it would be probably combine protocols from a number of the existing
standards, to address the different functions that need federating. I've
been trying to create a metamap of all the existing internet standards
and protocols here, input welcome:
https://metamaps.cc/maps/3152

**Comment by annando** [comment]
Currently there is no definition for nomadic identities in the AP standard.

**Comment by daita** [comment]
@jancborchardt status here ? do we close this ticket ?

Personally, I am not a fan of adding another protocol to the app; AP is good enough.

**Comment by annando** [comment]
From my outside position I would vote for closing this issue. But I suggest to be open for upcoming (non standard) extensions to the AP standard, when they do make sense for NC.
